pkg/operator/vclib/connection.go

```diff
@@ -24,7 +30,76 @@ type VSphereConnection struct {
 	Hostname   string
 	Port       string
 	Insecure   bool
-	Config     *vsphere.VSphereConfig
+	Config     *VSphereConfig
+}
+
+// VSphereConfig contains configuration for cloud provider.  It wraps the legacy version and the newer upstream version
+// with yaml support
+type VSphereConfig struct {
+	Config       *vsphere.Config
+	LegacyConfig *legacy.VSphereConfig
+}
+
+// LoadConfig load the desired config into this config object.
+func (c *VSphereConfig) LoadConfig(data string) error {
+	var err error
+	c.Config, err = vsphere.ReadConfig([]byte(data))
+	if err != nil {
+		return err
+	}
```

Comment:
I was just testing this and this code currently always throws:

```
I0726 20:00:55.719681       1 config.go:272] ReadConfig INI succeeded. INI-based cloud-config is deprecated and will be removed in 2.0. Please use YAML based cloud-config.
```

This will always be returned for upgraded clusters, since nothing is there to migrate these INI based configs to YAMLs. So this is one of the things we talked about in the call. Do we want to write a ticket or something for this?

Reply:
I am thinking we should rather check if we are dealing with YAML or ini and then load one or the other.

Reply:
The library we are using will check if YAML and if that fails, it will go back to INI.  The logging for this library is excessive, but we also load this config a lot when we check.  It would be nice to limit how often we are generating this config object.

Another alternative is if multi vCenter support is disabled (looking at the featuregate which when GA will always be true), we can load the old style config (INI).  This way its at least limited to output the messages when we are looking for the correct config.